src/tls.c

```diff
@@ -140,9 +141,11 @@ int ssl_init()
   }
   /* Load our own certificate and private key. Mandatory for acting as
      server, because we don't support anonymous ciphers by default. */
-  if (SSL_CTX_use_certificate_chain_file(ssl_ctx, tls_certfile) != 1)
+  if (SSL_CTX_use_certificate_chain_file(ssl_ctx, tls_certfile) != 1) {
+    ssl_files_loaded = 0;
```

Comment:
Shouldn't this also have some logic to change it back to 1 after a .rehash? Currently this looks like it would require a full restart.